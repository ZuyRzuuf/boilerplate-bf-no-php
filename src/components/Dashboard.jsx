import React from 'react';
import {Query} from 'react-apollo';

import Examples from '../../graphql/client/queries/examples.graphql';

const Dashboard = () => {
    return (
        <>
            <div>Dashboard</div>
            <Query query={Examples}>
                {({ loading, error, data }) => {
                    if (loading) return <p>Loading...</p>;

                    if (error) return <p>Error :(</p>;

                    return data.examples.map(({ uuid, name }) => (
                        <div key={uuid}>
                            <p>
                                {`${uuid}: ${name}`}
                            </p>
                        </div>
                    ));
                }}
            </Query>
        </>
    );
};

export default Dashboard;
