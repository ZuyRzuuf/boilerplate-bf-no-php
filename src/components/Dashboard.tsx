import * as React from 'react';
// import {Query} from 'react-apollo';

// import Examples from '../../graphql/client/queries/examples.graphql';

const Dashboard = () => {
    return (
        <>
            <div>Dashboard</div>
        </>
    );
};

export default Dashboard;
