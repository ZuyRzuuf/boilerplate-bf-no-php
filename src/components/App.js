import React from 'react';

import Header from "./Header.jsx";
import Dashboard from "./Dashboard.jsx";

class App extends React.Component {
    render() {
        return (
            <div>
                <Header/>
                <Dashboard/>
            </div>
        );
    }
}

export default App;
