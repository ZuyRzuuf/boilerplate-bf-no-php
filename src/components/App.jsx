import React from 'react';
import {
    BrowserRouter as Router,
    Route
} from 'react-router-dom';

import Header from './Header';
import Dashboard from './Dashboard';
import NotFound from './NotFound';
import {Switch} from 'react-router';

const App = () => {
    return (
        <Router>
            <>
                <Header />
                <Switch>
                    <Route component={Dashboard} exact path="/" />
                    <Route component={NotFound} />
                </Switch>
            </>
        </Router>
    );
};

export default App;
