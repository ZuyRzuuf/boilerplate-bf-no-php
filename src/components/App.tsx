import * as React from "react";
// import {
//     BrowserRouter as Router,
//     Route
// } from 'react-router-dom';
// import {Switch} from 'react-router';

import Header from './Header';
import Dashboard from './Dashboard';
// import NotFound from './NotFound';

const App = () => {
    return (
        <>
            <Header />
            <Dashboard />
        </>
    );
};

export default App;
