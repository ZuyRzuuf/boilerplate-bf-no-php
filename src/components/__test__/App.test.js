import React from 'react';
import { MemoryRouter } from 'react-router';
import { mount } from 'enzyme';

import App from '../App';
import Dashboard from '../Dashboard';
import NotFound from '../NotFound';

describe('A suite', function() {
    test('renders isolated component correctly', () => {
        const wrapper = shallow(
            <App />
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('renders components tree correctly', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={[ '/' ]} initialIndex={0}>
                <App />
            </MemoryRouter>
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('valid path should redirect to Dashboard and renders HTML tree correctly', () => {
        const html = render(
            <MemoryRouter initialEntries={[ '/' ]} initialIndex={0}>
                <App />
            </MemoryRouter>
        );
        expect(html).toMatchSnapshot();
    });

    test('valid path should redirect to Dashboard', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={[ '/' ]}>
                <App />
            </MemoryRouter>
        );
        expect(wrapper.find(Dashboard)).toHaveLength(1);
        expect(wrapper.find(NotFound)).toHaveLength(0);
    });

    test('invalid path should redirect to NotFound and renders HTML tree correctly', () => {
        const html = render(
            <MemoryRouter initialEntries={[ '/nonsense' ]}>
                <App />
            </MemoryRouter>
        );
        expect(html).toMatchSnapshot();
    });

    test('invalid path should redirect to NotFound', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={[ '/nonsense' ]}>
                <App />
            </MemoryRouter>
        );
        expect(wrapper.find(Dashboard)).toHaveLength(0);
        expect(wrapper.find(NotFound)).toHaveLength(1);
    });
});
