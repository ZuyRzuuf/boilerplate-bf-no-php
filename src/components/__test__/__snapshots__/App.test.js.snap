// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A suite invalid path should redirect to NotFound and renders HTML tree correctly 1`] = `
<div>
  <div>
    Backend/Frontend boilerplate without PHP
  </div>
  <div>
    NotFound
  </div>
</div>
`;

exports[`A suite renders components tree correctly 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      "/",
    ]
  }
  initialIndex={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "891bob",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "891bob",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <App>
      <Component>
        <div>
          <Header>
            <div>
              Backend/Frontend boilerplate without PHP
            </div>
          </Header>
          <Switch>
            <Route
              component={[Function]}
              computedMatch={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
              exact={true}
              location={
                Object {
                  "hash": "",
                  "key": "891bob",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              path="/"
            >
              <Dashboard
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "canGo": [Function],
                    "createHref": [Function],
                    "entries": Array [
                      Object {
                        "hash": "",
                        "key": "891bob",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      },
                    ],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "index": 0,
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "key": "891bob",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "key": "891bob",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  }
                }
                match={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
              >
                <div>
                  Dashboard
                </div>
              </Dashboard>
            </Route>
          </Switch>
        </div>
      </Component>
    </App>
  </Router>
</MemoryRouter>
`;

exports[`A suite renders isolated component correctly 1`] = `
<Component>
  <Header />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</Component>
`;

exports[`A suite valid path should redirect to Dashboard and renders HTML tree correctly 1`] = `
<div>
  <div>
    Backend/Frontend boilerplate without PHP
  </div>
  <div>
    Dashboard
  </div>
</div>
`;
